node/cli/src/service.rs

```diff
@@ -80,17 +80,19 @@ construct_service_factory! {
 				FullComponents::<Factory>::new(config, executor) },
 		AuthoritySetup = {
 			|service: Self::FullService, executor: TaskExecutor, key: Arc<Pair>| {
+				let (block_import, link_half) = service.config().custom.grandpa_import_setup.as_ref().take()
```

Comment:
Why do you use `as_ref()` and than `clone()` at the end? If you already `take()` the `Option`, you could just drop `as_ref` and `clone`.

Reply:
I think the original cloned wanted to take the object from the config, this only works if `service` is `mut` (hence the `as_ref()`). Unfortunately rustc seems to be ignoring the `mut` quantifier that I add to the `service` bindings.